cmd/collector/main.go

```diff
@@ -95,3 +106,43 @@ func logOptions(loggingConfigPath *string) ([]zap.Option, error) {
 
 	return l.Options()
 }
+
+func checkManagerConfig(configPath *string) error {
+	if _, err := os.Stat(*configPath); err == nil {
+		// manager config file exists, return and verify
+		return err
+
+	} else if errors.Is(err, os.ErrNotExist) {
+		// manager.ymal file does *not* exist, create file using env variables
+		newConfig := &opamp.Config{}
+		if ep, ok := os.LookupEnv(endpoint); ok {
+			newConfig.Endpoint = ep
+
+			var ai string
+			if ai, ok = os.LookupEnv(agentID); !ok {
+				ai = uuid.New().String()
+			}
+			newConfig.AgentID = ai
+
+			if sk, ok := os.LookupEnv(secretKey); ok {
+				newConfig.SecretKey = &sk
+			}
+
+			if label, ok := os.LookupEnv(labels); ok {
+				newConfig.Labels = &label
+			}
+
+			var data []byte
+			if data, err = yaml.Marshal(newConfig); err != nil {
+				return err
+			}
+			// write data to a manager.yaml file, with 0777 file permission
+			if err := os.WriteFile(*configPath, data, 0777); err != nil {
+				return err
```

Comment:
Same here something like `fmt.Errorf("failed to write config file created from ENVs: %w", err)`